mlflow/models/model.py

```diff
@@ -202,6 +208,11 @@ def to_json(self):
 
     def save(self, path):
         """Write the model as a local YAML file."""
+
+        # Saved model must have mlflow_version set
```

Comment:
This is set in the constructor. Any reason to validate here?

Reply:
but I don't like that option either. My view is that model loaded from an older file must have mlflow_version set to None but what happens when it saved - that's the decision that is needed. Setting a value on save creates a side effect but not doing so is against the new save approach. I thoughts?

Reply:
Yes, this business of loading a model from an older version file then saving it isn't straightforward. Arguably, if the model is saved into the same file as It was loaded from then I would probably not expect a new attribute to appear there (e.g. not to break some other tooling that may depend on that location). In contrast, if the model is saved to a different file that it was loaded from then perhaps adding a new attribute would be appropriate. Unfortunately, we don't capture the information about the file location on load so implementing this would not be possible without adding that info on load which sounds more like a different ticket/issue. 

How about leaving it to the user to explicitly set the mlflow_version if they load the model from an older version? And if they don't set the value then the attribute is not saved at all?

Reply:
we don't persist the loaded location on load, but the classmethod `load()` uses the uri as it's singular argument. Setting an instance param of that location could give you the information needed to enforce that backwards compatibility. 
I'll bring it up at standup and see what the team thinks about this behavior for older versions. Stay tuned!

Reply:
Cool, thanks, let me know what the team thinks. In the meantime, I'm going to submit a version that does not change the attribute when model is loaded from the older file

Reply:
We have a unanimous decision: "Let's not apply any additional information to a pre-mlflow-version-attribute to models that are loaded (or saved). So long as all new models receive this and handling a missing value in the artifact to maintain backwards compatibility are maintained, we're good to go."

Reply:
Just to verify, sounds like this design matches my current implementation, please confirm or correct my understanding. Currently, I store `mlflow_version=None` if the model is loaded from the pre-mlflow-version-attribute file and don't save into file the `mlflow_version` attribute at all if a save is called on that model.